app/controllers/web_rake/tasks_controller.rb

```diff
@@ -80,81 +79,53 @@ def ensure_environment_loaded
     end
 
     def custom_tasks
-      # First, get the names of tasks defined in lib/tasks files
-      custom_task_names = extract_task_names_from_files
-
-      # Always include db:seed task
-      custom_task_names << 'db:seed'
-
       # Ensure all tasks are loaded
       Rails.application.load_tasks
 
-      # Get only the tasks that match our custom task names
-      custom_task_names.filter_map do |name|
-        Rake::Task[name] rescue nil
-      end.select(&:actions)
-    end
-
-    def extract_task_names_from_files
-      task_names = []
-
-      # Parse each rake file to extract task names
-      Dir.glob(Rails.root.join('lib/tasks/**/*.rake')).each do |file|
-        content = File.read(file)
-
-        # Match various task definition patterns
-
-        # Pattern for: task({ :sample_data => :environment })
-        content.scan(/task\s*\(\s*\{\s*:([a-zA-Z_][a-zA-Z0-9_]*)\s*=>/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task({ sample_data: :environment }) - Ruby 1.9+ syntax with parens and braces
-        content.scan(/task\s*\(\s*\{\s*([a-zA-Z_][a-zA-Z0-9_]*)\s*:/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task :sample_data => :environment
-        content.scan(/task\s+:([a-zA-Z_][a-zA-Z0-9_]*)\s*=>/).each do |match|
-          task_names << match[0]
-        end
+      tasks = []
 
-        # Pattern for: task "sample_data" => :environment
-        content.scan(/task\s+["']([a-zA-Z_][a-zA-Z0-9_]*)["']\s*=>/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task sample_data: :environment (Ruby 1.9+ syntax)
-        content.scan(/task\s+([a-zA-Z_][a-zA-Z0-9_]*)\s*:/).each do |match|
-          task_names << match[0]
-        end
+      # Always include db:seed
+      if (seed_task = Rake::Task['db:seed'] rescue nil)
+        tasks << seed_task
+      end
 
-        # Pattern for: task(:sample_data)
-        content.scan(/task\s*\(\s*:([a-zA-Z_][a-zA-Z0-9_]*)\s*\)/).each do |match|
-          task_names << match[0]
-        end
+      # Find tasks defined in lib/tasks by checking their source location
+      Rake.application.tasks.each do |task|
+        next if task.actions.blank?
+        next if task.name == 'db:seed' # Already added above
 
-        # Pattern for: desc "..." \n task :name
-        content.scan(/desc\s+.*?\n\s*task\s+:([a-zA-Z_][a-zA-Z0-9_]*)/).each do |match|
-          task_names << match[0]
+        if task_from_lib_tasks?(task)
```

Comment:
```suggestion
        if task_from_lib_tasks?(task) || task.name == "db:seed"
```

if we do this, I think we could remove the "Always include db:seed" if statement and `next if task.name == 'db:seed'` above.